File: Source/Components/HmacResult.cs
namespace HmacManagement.Components;

public class HmacResult
{
    public required string Policy { get; init; }
    public required string HeaderScheme { get; init; }
    public Hmac? Hmac { get; init; } = default;
    public bool IsSuccess { get; init; } = false;
    public DateTime DateGenerated { get; } = DateTime.UtcNow;
}

File: src/Scripts/HmacManager/components/hmac-result.ts
import { Hmac } from "./hmac.js";

export class HmacResult {
    policy: string;
    scheme: string;
    hmac: Hmac;
    isSuccess: boolean;
    dateGenerated: Date;
    information: string;

    constructor(hmac, isSuccess) {
        this.hmac = hmac
        this.isSuccess = isSuccess
    }
}